### PR TITLE
feat(docs): define docs map schema and doc kind taxonomy

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Creates Zod schemas for the docs map manifest in `packages/docs/src/core/docs-map-schema.ts`
- Defines `DocsMap` (top-level manifest), `DocsMapEntry` (individual doc metadata), and `DocKind` taxonomy
- Doc kinds: `readme`, `guide`, `reference`, `architecture`, `release`, `convention`, `deep`, `generated`
- Output location: `.outfitter/docs-map.json` alongside existing `surface.json`
- Full test coverage for schema validation and edge cases

## Test plan

- [ ] Schema validates well-formed docs map entries
- [ ] Schema rejects malformed entries with clear errors
- [ ] `DocKind` enum covers all expected document categories
- [ ] Exports available via `@outfitter/docs/core`

Closes OS-277

Closes: OS-277

<h3>Greptile Summary</h3>

Introduces foundational schema infrastructure for the docs map manifest (`.outfitter/docs-map.json`). Defines three core schemas: `DocKindSchema` (8-member taxonomy: readme, guide, reference, architecture, release, convention, deep, generated), `DocsMapEntrySchema` (individual doc metadata with id, kind, title, paths, optional package/tags), and `DocsMapSchema` (top-level manifest with timestamp, generator, and entries array).

Implementation follows established patterns:
- Uses explicit `ZodType<T>` annotations matching `apps/outfitter/src/manifest.ts` style (addressed in previous review thread)
- Comprehensive TSDoc with usage examples for each schema
- 23 passing tests covering validation, edge cases, defaults, and type inference
- Proper exports via both main entry point and direct `@outfitter/docs/core/docs-map-schema` path
- Adds `zod` as catalog dependency alongside existing `better-result` and `commander`

No logical issues, security concerns, or architectural problems identified. The PR establishes a clean foundation for docs indexing alongside the existing `surface.json` schema drift detection system.

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no concerns
- Schema-only PR with full test coverage (23/23 passing), proper type annotations following codebase conventions, no runtime logic, no breaking changes, and addresses previous review feedback about explicit ZodType annotations
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/docs/src/core/docs-map-schema.ts | Defines Zod schemas for docs map manifest with explicit type annotations, comprehensive TSDoc, and proper taxonomy structure |
| packages/docs/src/__tests__/docs-map-schema.test.ts | Comprehensive test suite with 23 passing tests covering validation, edge cases, and type inference |
| packages/docs/src/index.ts | Adds proper exports for new schemas and types following existing package conventions |
| packages/docs/package.json | Adds zod dependency and new export path for direct schema access |

</details>

</details>

<sub>Last reviewed commit: 5f16e1e</sub>

### CODE REVIEW

**Comment by galligan** [comment]
@greptileai review

Focus areas:
- **Schema design**: `DocsMap`, `DocsMapEntry`, and `DocKind` schemas should be extensible without breaking changes — check for optional fields and future-proofing
- **Taxonomy completeness**: The `DocKind` enum (`readme`, `guide`, `reference`, `architecture`, `release`, `convention`, `deep`, `generated`) should cover all current doc types without overlap
- **Zod v4 usage**: This codebase uses Zod v4 — verify the schema definitions use correct Zod v4 syntax (not v3)
- **Export surface**: Schemas should be exported from `@outfitter/docs/core` for consumers

File: packages/docs/src/core/docs-map-schema.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ *   generator: "@outfitter/docs@0.1.2",
+ *   entries: [],
+ * });
+ * ```
+ */
+export const DocsMapSchema = z.object({
+  /** Optional JSON Schema URI for editor validation. */
+  $schema: z.string().optional(),
+  /** ISO-8601 timestamp of when the map was generated. */
+  generatedAt: z.string(),
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-schema.ts]: Addressed in subsequent commits on this stack; this thread is now outdated relative to the current diff. Resolving.
